### PR TITLE
Add ingress provider labels to contour resources

### PR DESCRIPTION
**Pull request by shashwathi**
Add bash function to inject ingress provider labels to contour config resources

Signed-off-by: Shash Reddy <shashwathireddy@gmail.com>

cc @andrew-su 

/assign @tcnghia 
/assign @mattmoor

### CODE REVIEW

File: hack/labels.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+metadata:
+  labels:
+    networking.knative.dev/ingress-provider: contour
```
Comment (mattmoor) [inline hack/labels.yaml:4]: Add trailing newline:
```suggestion
    networking.knative.dev/ingress-provider: contour

```

File: hack/update-deps.sh [hunk 2]
```diff
@@ -158,4 +177,5 @@ KO_DOCKER_REPO=ko.local ko resolve -f ./vendor/github.com/projectcontour/contour
   | rewrite_contour_namespace contour-external \
   | configure_leader_election contour-external \
   | rewrite_serve_args contour-external \
-  | rewrite_image | rewrite_command | disable_hostport >> config/contour/external.yaml
+  | rewrite_image | rewrite_command | disable_hostport \
+  | add_ingress_provider_labels >> config/contour/external.yaml
```
Comment (mattmoor) [inline hack/update-deps.sh:181]: Add trailing newline:
```suggestion
  | add_ingress_provider_labels >> config/contour/external.yaml

```

**Comment by andrew-su** [comment]
@googlebot I consent.

File: hack/update-deps.sh (hunk 2 above)
Comment (mattmoor) [inline hack/update-deps.sh:181]: Add trailing newline:
```suggestion
  | add_ingress_provider_labels >> config/contour/external.yaml

```

Comment (mattmoor) [inline hack/update-deps.sh:181]: Add trailing newline:
```suggestion
  | add_ingress_provider_labels >> config/contour/external.yaml

```

Comment (mattmoor) [inline hack/update-deps.sh:181]: Add trailing newline:
```suggestion
  | add_ingress_provider_labels >> config/contour/external.yaml

```

File: hack/tools.go [hunk 3]
```diff
@@ -34,4 +34,5 @@ import (
```
Comment (mattmoor) [inline hack/tools.go:34]: Format Go code:
```suggestion

	_ "github.com/mikefarah/yq/v3"
```

File: hack/tools.go [hunk 4]
```diff
@@ -34,4 +34,5 @@ import (
 
 	_ "github.com/projectcontour/contour/cmd/contour"
 	_ "github.com/projectcontour/contour/examples/contour"
+	_ "github.com/mikefarah/yq/v3"
```
Comment (mattmoor) [inline hack/tools.go:37]: Format Go code:
```suggestion

```

**Comment by tcnghia** [comment]
`LGTM analysis: JavaScript` is failing, do we have any JavaScript? 

/lgtm
/approve

**Pull request merged**